src/components/Table/README.md

```diff
@@ -0,0 +1,172 @@
+            await fetchLandings()
+          }, [fetchLandings, refresh]) <- dzięki dodaniu parametru refresh udostępnionego z hooka możemy manulanie triggerować odświeżenie tabeli
+
+        return <Table />
+   }
+```
+
+Poszczególne parametry hooka useTable będą wytłumaczone w następnym rozdziale.
+
+## Propsy
+
+- `name(wymagany)` - Nazwa tabeli, wykorzystywane do zapamiętywania parametru
+  page size w localstorage
+- `data(wymagany)` - Tablica z danymi do wylistowania, każdy element składa się
+  z obiektu, w którym klucz `identifier` jest wymagany, reszta parametrów jest
+  dowolna i ich definicję (jeśli korzystamy z typescripta) podajemy jako
+  parametr do typu generycznego tzn. `useTable()` przykład poprawnego formatu
+  itemu:
+  ```
+  {
+      identifier: string | number; <- ten klucz jest wymagany, reszta tego obiektu to pola potrzebne przy dalszych etapach
+      name: string; <- (przykład) na jego podstawie renderowana będzie wartość komórki, więcej info w następnym podpunkcie
+      url: string;
+      ...rest
+  }[]
+  ```
+- `columns(wymagany)` - Tablica z obiektami, które definiują, z jakich kolumn
+  będzie się składać wyrenderowana tabelka, w tym miejscu do tablicy możemy
+  wprowadzić dwa rodzaje komumn: bazującą na `accessor key` oraz na customowym
+  renderowaniu komórki, przykład
+
+  ```
+  type Width = number | `${number}%`
+
+  export interface ColumnAccessor<Item> {
+    identifier: string
+    accessor: keyof Item <- Weryfikuje czy klucz, który podamy faktycznie istnieje w pojedynczym elemencie w tablicy data
+    width?: Width <- Opcjonalne, customowa szerokość pola podawana w px lub %
+    header?: string <- Nazwa wyświetlana w headerze
+  }
+
+  export interface CustomColumn<Item> {
+    identifier: string
+    render: (item: Item, handleRefresh: () => void) => ReactNode <- funkcja, w której wyrenderować możemy customową komórkę, udostępnia nam wszystkie dane pojedynczego elementu w tablicy data i callback do odświeżenia tablicy
+    width?: Width
+    header?: string
+  }
+
+  const columns = [
+          {
+              header: 'column based on accessor key',
+              identifier: 'accessor-column-identifier',
+              accessor: 'name', <- Tutaj możemy wprowadzić tylko i wyłącznie klucz, który istnieje na pojedynczym elemencie w tabeli data, w tym przypadku jest to klucz 'name', który podałem wyżej ts będzie to weryfikował
+              width: '100%' <- Wartość liczbowa (px) lub procentowa, type Width = number | `${number}%`
+          },
+          {
+              header: 'info',
+              identifier: 'info',
+              render: info => <LandingInfo {...info} />,
+              width: '100%'
+          },
+      ]
+  ```
+
+  Możemy mieszać ze sobą typy kolumn na poziomie tablicy, natomiast nie możemy
+  zdefiniować jednocześnie accessor i render w jednym obiekcie, ts rzuci błędem.
+
+- `i18n(wymagany)` - props wymagany składa się z 3 kluczy z tłumaczeniami,
```

Comment:
istnieją takie tabele, które nie mają z założenia paginacji, ani headera, ani checkboxów i nie potrzebują żadnego z tych tłumaczeń

Reply:
Teoretycznie można to zrobić ale nie wiem czy warto, musiałbym dość pokomplikowac typy aby te tłumaczenia były wymagane, gdy np jest paganiacja itd. LUB droga na skróty, zrobić ten props opcjonalny i na tłumaczeniach dać fallback z kluczem np. też jest to jakieś rozwiązanie aby w szczególnych przypadkach tego nie podawać

Reply:
ja bym zrobiła po prostu jako opcjonalny + jakiś podstawowy fallback w tłumaczeniach, zazwyczaj to jest dokładnie to samo tłumaczenie w każdej tabeli :)